src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1241,6 +1243,69 @@ class spell_pri_vampiric_touch : public SpellScriptLoader
         }
 };
 
+// 1706 - Levitate
+class spell_pri_levitate : public SpellScriptLoader
+{
+public:
+    spell_pri_levitate() : SpellScriptLoader("spell_pri_levitate") { }
+
+    class spell_pri_levitate_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_pri_levitate_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PRIEST_LEVITATE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_PRIEST_LEVITATE_EFFECT))
+                return false;
+            return true;
+        }
+        
+        bool Load() override
+        {
+            return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+        }
+        
+        SpellCastResult CheckCast()
+        {
+            Player* caster = GetCaster()->ToPlayer();
+
+            if (Unit* target = GetExplTargetUnit())
+            {
+                if (!caster->IsInRaidWith(target))
+                    return SPELL_FAILED_BAD_TARGETS;
+
+                if (!target->IsAlive())
+                    return SPELL_FAILED_DONT_REPORT;
+            }
+
+            return SPELL_CAST_OK;
+        }
+        
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetCaster())
+            {
+                if (Unit* target = GetHitUnit()) 
```

Comment:
GetCaster cant be null in SpellScripts
And in this hook, GetHitUnit cant be null too (can be null only if target is a object)